mne/decoding/transformer.py

```diff
@@ -245,6 +245,97 @@ def inverse_transform(self, X, y=None):
         return X.reshape(-1, self.n_channels, self.n_times)
 
 
+class Vectorizer(TransformerMixin):
+    """Transforms n-dimensional array into 2D array of n_sample by n_features.
+
+    This class reshapes an n-dimensional array into an n_sample * n_features
+    array, usable by the estimators and transformers of scikit-learn.
+
+    Examples
+    --------
+    clf = make_pipeline(SpatialFilter(), XdawnTransformer(), Vectorizer(),
+                        LogisticRegression())
+
+    Attributes
+    ----------
+    ``shape_`` : array
+         Stores the original dimension of data.
+    """
+
+    def fit(self, X, y=None):
+        """Stores the shape of X.
+
+        Parameters
+        ----------
+        X : array-like
+            The data to fit. Can be, for example a list, or an array of at
+            least 2d. The first dimension must be of shape (n_samples).
```

Comment:
yes, althought vectorizer will be generic, and thus applicable to epoched, or raw data. @kaichogami  may be write 'of length n_samples, where samples are the independent samples used by the estimator (e.g. n_epochs for epoched data).